test/unit/webui/components/search.spec.js

```diff
@@ -0,0 +1,204 @@
+/**
+ * @flow
+ * @prettier
+ */
+
+import React from 'react';
+import { mount } from 'enzyme';
+
+import Search from '../../../../src/webui/components/Search/index';
+
+const SEARCH_FILE_PATH = '../../../../src/webui/components/Search/index';
+
+// Global mocks
+const event = {
+  stopPropagation: jest.fn(),
+};
+
+window.location.assign = jest.fn();
+
+describe('<Search /> component test', () => {
+  let wrapper;
+  beforeEach(() => {
+    wrapper = mount(<Search />);
+  });
+
+  test('should load the component in default state', () => {
+    expect(wrapper.html()).toMatchSnapshot();
+  });
+
+  test('onBlur: should cancel all search requests', async () => {
+    const { onBlur, requestList } = wrapper.instance();
+    const spy = jest.spyOn(wrapper.instance(), 'cancelAllSearchRequests');
+    // add one request
+    const request = {
+      abort: jest.fn(),
+    };
+    wrapper.instance().requestList = [request];
+
+    onBlur(event);
+
+    expect(event.stopPropagation).toHaveBeenCalled();
+    expect(wrapper.state('error')).toBeFalsy();
+    expect(wrapper.state('loaded')).toBeFalsy();
+    expect(wrapper.state('loading')).toBeFalsy();
+    expect(spy).toHaveBeenCalled();
+    expect(requestList).toEqual([]);
+  });
+
+  test('handleSearch: should set value in state when value is defined', () => {
```

Comment:
I think this title needs a bit more context, what I read in the code is your aim to test the `loading` state process while you type. What do you think?

Reply:
Agree !